domain/src/main/java/com/ftw/domain/usecase/login/LoginUseCaseImpl.kt

```diff
@@ -1,19 +1,22 @@
 package com.ftw.domain.usecase.login
 
+import com.ftw.domain.entity.KakaoToken
 import com.ftw.domain.repository.login.LoginRepository
 import com.ftw.domain.repository.token.TokenRepository
-import kotlinx.coroutines.flow.Flow
-import kotlinx.coroutines.flow.map
 
 class LoginUseCaseImpl(
     private val tokenRepository: TokenRepository,
     private val loginRepository: LoginRepository
 ) : LoginUseCase {
-    override suspend fun invoke(): Flow<Result<String>> {
-        val tokenResult = tokenRepository.getUserToken()
-        return tokenResult.map { result ->
-            result.getOrNull()?.let { token -> loginRepository.login(token) }
-                ?: Result.failure(IllegalAccessException("Login is failed"))
+
+    //1. 서버에 카카오 토큰을 넘겨 서버에 저장
+    //2. 받아올 jwt토큰을 로컬에 저장
+    override suspend fun signUp(kakaoToken: KakaoToken):Result<Boolean>{
+        val serverResult = loginRepository.setKaKaoToken(kakaoToken)
+        serverResult.getOrNull()?.let {
+            tokenRepository.setUserToken(it)
+            return Result.success(true)
         }
+        return Result.failure(IllegalAccessException("signUp 실패"))
```

Comment:
setKakaoToken 의 return 값이 null 이면 미리 return 하면 어때?
```suggestion
        val token = loginRepository.setKaKaoToken(kakaoToken).getOrNull() ?: return Result.failure(IllegalAccessException("signUp 실패"))
        tokenRepository.setUserToken(token)
        return Result.success(true)
```

Reply:
우와! 엄청 깔끔해요!